File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/calliope2017/actors_singlemotor_and_motionkit_without_pins.cpp
#define _GNU_SOURCE

#include "MicroBit.h"
#include "NEPODefs.h"
#include <list>
#include <array>
#include <stdlib.h>
MicroBit _uBit;

void ____move();

void ____wait();



double ___off;
double ___max;
double ___negative;

int main()
{
    _uBit.init();
    ___off = 0;
    ___max = 100;
    ___negative = 0;

    ____move();
    release_fiber();
}

void ____move() {
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("m1 On"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("m1 On")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.soundmotor.motorOn(___max);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("M1 float"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("M1 float")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.soundmotor.motorCoast();
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("M1 Negative"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("M1 Negative")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.soundmotor.motorOn(___negative);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("Stop"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("Stop")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.soundmotor.motorAOff();
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("M1 On"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("M1 On")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.soundmotor.motorOn(___max);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("Brake"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("Brake")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.soundmotor.motorBreak();
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("M1 On"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("M1 On")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.soundmotor.motorOn(___max);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("Sleep"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("Sleep")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.soundmotor.motorSleep();
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("-----------------------Motion kit:-------------------"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("-----------------------Motion kit:-------------------")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("left FW"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("left FW")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P8.setServoValue(180);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("left BW"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("left BW")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P8.setServoValue(0);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("left Off"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("left Off")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P8.setAnalogValue(0);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("right FW"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("right FW")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P2.setServoValue(0);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("right BW"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("right BW")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P2.setServoValue(180);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("right OFF"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("right OFF")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P2.setAnalogValue(0);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("Turn right"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("Turn right")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P2.setServoValue(180);
    _uBit.io.P8.setServoValue(180);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("Turn left"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("Turn left")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P2.setServoValue(0);
    _uBit.io.P8.setServoValue(0);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("Forwards"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("Forwards")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P2.setServoValue(0);
    _uBit.io.P8.setServoValue(180);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("Backwards"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("Backwards")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P2.setServoValue(180);
    _uBit.io.P8.setServoValue(0);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("STOP"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("STOP")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P2.setAnalogValue(0);
    _uBit.io.P8.setAnalogValue(0);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("Only left on"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("Only left on")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P2.setAnalogValue(0);
    _uBit.io.P8.setServoValue(180);
    ____wait();
    _uBit.serial.setTxBufferSize(ManagedString((ManagedString("Only right on"))).length() + 2);
    _uBit.serial.send(ManagedString(ManagedString("Only right on")) + "\r\n", MicroBitSerialMode::ASYNC);
    _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    _uBit.io.P2.setServoValue(0);
    _uBit.io.P8.setAnalogValue(0);
}

void ____wait() {
    while (true) {
        if ( _uBit.buttonA.isPressed() == true ) {
            break;
        }
        _uBit.sleep(_ITERATION_SLEEP_TIMEOUT);
    }
    _uBit.sleep(500);
}

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/calliopev3/actors_singlemotor_and_motionkit_without_pins.py
import calliopemini
import random
import math

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

timer1 = calliopemini.running_time()
calliopemini.pin_A1_RX.set_analog_period(20)
calliopemini.pin_A1_TX.set_analog_period(20)


___off = 0
___max = 100
___negative = 0
def ____move():
    global timer1, ___off, ___max, ___negative
    print("m1 On")
    set_motor("A", ___max)
    ____wait()
    print("M1 float")
    set_motor("A", 0)
    ____wait()
    print("M1 Negative")
    set_motor("A", ___negative)
    ____wait()
    print("Stop")
    set_motor("A", 0)
    ____wait()
    print("M1 On")
    set_motor("A", ___max)
    ____wait()
    print("Brake")
    set_motor("A", -0.01)
    ____wait()
    print("M1 On")
    set_motor("A", ___max)
    ____wait()
    print("Sleep")
    set_motor("A", 0)
    ____wait()
    print("-----------------------Motion kit:-------------------")
    print("left FW")
    calliopemini.pin_A1_TX.write_analog(servo_get_angle(180))
    ____wait()
    print("left BW")
    calliopemini.pin_A1_TX.write_analog(servo_get_angle(0))
    ____wait()
    print("left Off")
    calliopemini.pin_A1_TX.write_analog(0)
    ____wait()
    print("right FW")
    calliopemini.pin_A1_RX.write_analog(servo_get_angle(0))
    ____wait()
    print("right BW")
    calliopemini.pin_A1_RX.write_analog(servo_get_angle(180))
    ____wait()
    print("right OFF")
    calliopemini.pin_A1_RX.write_analog(0)
    ____wait()
    print("Turn right")
    calliopemini.pin_A1_RX.write_analog(servo_get_angle(180))
    calliopemini.pin_A1_TX.write_analog(servo_get_angle(180))
    ____wait()
    print("Turn left")
    calliopemini.pin_A1_RX.write_analog(servo_get_angle(0))
    calliopemini.pin_A1_TX.write_analog(servo_get_angle(0))
    ____wait()
    print("Forwards")
    calliopemini.pin_A1_RX.write_analog(servo_get_angle(0))
    calliopemini.pin_A1_TX.write_analog(servo_get_angle(180))
    ____wait()
    print("Backwards")
    calliopemini.pin_A1_RX.write_analog(servo_get_angle(180))
    calliopemini.pin_A1_TX.write_analog(servo_get_angle(0))
    ____wait()
    print("STOP")
    calliopemini.pin_A1_RX.write_analog(0)
    calliopemini.pin_A1_TX.write_analog(0)
    ____wait()
    print("Only left on")
    calliopemini.pin_A1_RX.write_analog(0)
    calliopemini.pin_A1_TX.write_analog(servo_get_angle(180))
    ____wait()
    print("Only right on")
    calliopemini.pin_A1_RX.write_analog(servo_get_angle(0))
    calliopemini.pin_A1_TX.write_analog(0)

def ____wait():
    global timer1, ___off, ___max, ___negative
    while True:
        if calliopemini.button_a.is_pressed() == True:
            break
    calliopemini.sleep(500)

def run():
    global timer1, ___off, ___max, ___negative
    ____move()

def main():
    try:
        run()
    except Exception as e:
        raise

def servo_get_angle(angle):
    if (angle < 0): 
        angle = 0
    if (angle > 180): 
        angle = 180
    return round(0.55 * angle) + 25

def set_motor(port, speed):
    digit = 0
    if (speed < 0):
        digit = 1
        speed *= -1
    speed = int(speed * 10.23)
    calliopemini.pin_M_MODE.write_digital(1)
    if (port == "A"):
        calliopemini.pin_M0_DIR.write_digital(digit)
        calliopemini.pin_M0_SPEED.write_analog(speed)
    else:
        calliopemini.pin_M1_DIR.write_digital(digit)
        calliopemini.pin_M1_SPEED.write_analog(speed)

if __name__ == "__main__":
    main()